Build failure: mwprocapture

### Steps To Reproduce

Version V1.3.0.4373 of mwprocapture (the Linux driver for the Magewell Pro Capture family) FTBFS when building against Linux 6.8 or newer.

Steps to reproduce the behavior:

- Add `hardware.mwProCapture.enable = true;` to your Nix configuration.
- Add `boot.kernelPackages = pkgs.linuxPackages_latest;` to your Nix configuration.
- `sudo nixos-rebuild switch`

### Build log

```
mwprocapture> Running phase: unpackPhase
mwprocapture> unpacking source archive /nix/store/70a7a2xkrc8mfxlin4jc8zkkblxpff9r-ProCaptureForLinux_4373.tar.gz
mwprocapture> source root is ProCaptureForLinux_4373
mwprocapture> setting SOURCE_DATE_EPOCH to timestamp 1688973555 of file ProCaptureForLinux_4373/src/ProCaptureLib_64.lib
mwprocapture> Running phase: patchPhase
mwprocapture> Running phase: updateAutotoolsGnuConfigScriptsPhase
mwprocapture> Running phase: configurePhase
mwprocapture> no configure script, doing nothing
mwprocapture> Running phase: buildPhase
mwprocapture> build flags: SHELL=/nix/store/r9h133c9m8f6jnlsqzwf89zg9w0w78s8-bash-5.2-p15/bin/bash KERNELDIR=/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/build
mwprocapture> cp ProCaptureLib_64.lib ProCaptureLib.o
mwprocapture> touch .ProCaptureLib.o.cmd
mwprocapture> make -C /nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/build M=/build/ProCaptureForLinux_4373/src modules
mwprocapture> make[1]: Entering directory '/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/build'
mwprocapture>   CC [M]  /build/ProCaptureForLinux_4373/src/sources/dma/mw-dma-mem.o
mwprocapture>   CC [M]  /build/ProCaptureForLinux_4373/src/sources/dma/mw-dma-user.o
mwprocapture>   CC [M]  /build/ProCaptureForLinux_4373/src/sources/dma/mw-dma-kernel.o
mwprocapture>   CC [M]  /build/ProCaptureForLinux_4373/src/sources/dma/mw-dma-phy.o
mwprocapture>   CC [M]  /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.o
mwprocapture> /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.c: In function 'os_strlcpy':
mwprocapture> /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.c:785:12: error: implicit declaration of function 'strlcpy'; did you mean 'strscpy'? [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
mwprocapture>   785 |     return strlcpy(dst, src, (size_t)size);
mwprocapture>       |            ^~~~~~~
mwprocapture>       |            strscpy
mwprocapture> cc1: some warnings being treated as errors
mwprocapture> make[3]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/scripts/Makefile.build:243: /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.o] Error 1
mwprocapture> make[2]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/Makefile:1921: /build/ProCaptureForLinux_4373/src] Error 2
mwprocapture> make[1]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/Makefile:240: __sub-make] Error 2
mwprocapture> make[1]: Leaving directory '/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/build'
mwprocapture> make: *** [Makefile:33: all] Error 2
error: builder for '/nix/store/dharaanszxvx0ssv91vagh2bwa9d6yc7-mwprocapture-1.3.0.4373-6.8.4.drv' failed with exit code 2;
       last 10 log lines:
       > /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.c:785:12: error: implicit declaration of function 'strlcpy'; did you mean 'strscpy'? [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
       >   785 |     return strlcpy(dst, src, (size_t)size);
       >       |            ^~~~~~~
       >       |            strscpy
       > cc1: some warnings being treated as errors
       > make[3]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/scripts/Makefile.build:243: /build/ProCaptureForLinux_4373/src/sources/ospi/ospi-linux.o] Error 1
       > make[2]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/Makefile:1921: /build/ProCaptureForLinux_4373/src] Error 2
       > make[1]: *** [/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/source/Makefile:240: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/4r9dpb5wgvx02f4b7ad70jkw44ajmwc8-linux-6.8.4-dev/lib/modules/6.8.4/build'
       > make: *** [Makefile:33: all] Error 2
       For full logs, run 'nix log /nix/store/dharaanszxvx0ssv91vagh2bwa9d6yc7-mwprocapture-1.3.0.4373-6.8.4.drv'.
error: 1 dependencies of derivation '/nix/store/34b7rj45nar1f7v2dryc89nva8m8jcaz-etc-modprobe.d-nixos.conf.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-nvidia-x11-535.129.03-6.8.4.drv-5/NVIDIA-Linux-x86_64-535.129.03/kernel': Directory not empty
error: 1 dependencies of derivation '/nix/store/32ngya44gc1vskzgpax17av77is5f28s-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fzy9y9hgx265b60h7vl5cillj4ryj4rs-linux-6.8.4-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qalg4r4rw3h63j81n9y90p901xxa802a-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b9gyi6y9ddxkxsxamlyjzchz3ps744rp-mwprocapture-1.3.0.4373-6.8.4_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/swslsjzyx26pdihy97pvhyd3sn1f73pb-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jp20ywssnz5w3a7ljrdrz7clszsw71fk-nixos-system-phasma--.drv' failed to build
```

### Additional context

Magewell has provided a revised driver update to address this issue. I am working on a pull request for nixpkgs.

### Notify maintainers

@MP2E

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.12, NixOS, 23.11 (Tapir), 23.11.20240406.e38d7cb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/wcg8s4srapl39ibwvhwiy5b7kvyjy7vq-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/mwprocapture/default.nix
+++ b/pkgs/os-specific/linux/mwprocapture/default.nix
@@ -12,12 +12,12 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "mwprocapture";
-  subVersion = "4373";
+  subVersion = "4390";
   version = "1.3.0.${subVersion}-${kernel.version}";
 
   src = fetchurl {
     url = "https://www.magewell.com/files/drivers/ProCaptureForLinux_${subVersion}.tar.gz";
-    sha256 = "sha256-/6q+6CTlgkHOgq1PF8dSPfl/xm/UFczr/AGkac2mXZ8=";
+    sha256 = "sha256-HOVAR9auc8ulENPLoI0scdCMZoSbDYkTaCLgZoFG7eU=";
   };
 
   nativeBuildInputs = kernel.moduleBuildDependencies;

